SquirrelPanel.m

```diff
@@ -993,8 +990,12 @@ - (void)showPreedit:(NSString *)preedit
         labelRange2 = NSMakeRange(NSNotFound, 0);
         labelFormat2 = nil;
         // fix label1, everything other than '%@' is label1
-        labelRange.location = 0;
-        labelRange.length = pureCandidateRange.location;
+        if (pureCandidateRange.location > 0) {
+          labelRange.location = 0;
+          labelRange.length = pureCandidateRange.location;
+        } else {
+          labelRange = NSMakeRange(NSNotFound, 0);
+        }
```

Comment:
The changes above seems to be taken together with part of the previous two commits.
Still haven't figured out what's going on. I'll move on to the easier code refactor first.

Reply:
Think of:
1. `%@`
2. `<%@>`
3. `%@ %c`
4. `<%@ %c>`